compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -28,54 +28,95 @@ import scala.collection.concurrent.TrieMap
 import scala.jdk.CollectionConverters._
 import scala.util.control.NonFatal
 
-object ScenarioServiceMain extends App {
+private final case class ScenarioServiceConfig(
+    maxInboundMessageSize: Int,
+    enableScenarios: Boolean,
+)
+
+private object ScenarioServiceConfig {
   // default to 128MB
-  val maxMessageSize = args.headOption.map(_.toInt).getOrElse(128 * 1024 * 1024)
-
-  // Needed for the akka Ledger bindings used by Daml Script.
-  val system = ActorSystem("ScriptService")
-  implicit val sequencer: ExecutionSequencerFactory =
-    new AkkaExecutionSequencerPool("ScriptServicePool")(system)
-  implicit val materializer: Materializer = Materializer(system)
-  implicit val ec: ExecutionContext = system.dispatcher
-
-  val server =
-    NettyServerBuilder
-      .forAddress(new InetSocketAddress(InetAddress.getLoopbackAddress, 0)) // any free port
-      .addService(new ScenarioService())
-      .maxInboundMessageSize(maxMessageSize)
-      .build
-  server.start()
-  // Print the allocated port for the client
-  println("PORT=" + server.getPort.toString)
-
-  // Bump up the log level
-  Logger.getLogger("io.grpc").setLevel(Level.ALL)
-
-  // Start a thread to watch stdin and terminate
-  // if it closes. This makes sure we do not leave
-  // this process running if the parent exits.
-  new Thread(new Runnable {
-    def run(): Unit = {
-      while (System.in.read >= 0) {}
-      System.err.println("ScenarioService: stdin closed, terminating server.")
-      server.shutdown()
-      system.terminate()
-      ()
-    }
-  }).start()
+  val DefaultMaxInboundMessageSize: Int = 128 * 1024 * 1024
+
+  private class OptionParser()
+      extends scopt.OptionParser[ScenarioServiceConfig]("scenario-service") {
+    head("scenario-service")
+
+    opt[Int]("max-inbound-message-size")
+      .action((x, c) => c.copy(maxInboundMessageSize = x))
+      .optional()
+      .text(
+        s"Optional max inbound message size in bytes. Defaults to ${DefaultMaxInboundMessageSize}."
+      )
+
+    opt[Boolean]("enable-scenarios")
+      .optional()
+      .action((x, c) => c.copy(enableScenarios = x))
+      .text(
+        "Enable/disable support for running scenarios. Defaults to true."
+      )
+  }
+
+  def parse(args: Array[String]): Option[ScenarioServiceConfig] =
```

Comment:
usually you’d put this in an `Object`. It’s really a static method.

Reply:
hm it's defined inside `private object ScenarioServiceConfig`, isn't that what you mean?

Reply:
oh sorry I misread the parentheses and thought it was defined within the class. all good :+1

Reply:
ah cool thanks :)